mutiny-server/src/routes.rs

```diff
@@ -0,0 +1,249 @@
+        .node_manager
+        .send_to_address(
+            address,
+            request.amount_sat,
+            label,
+            request.fee_rate_sat_byte,
+        )
+        .await
+        .map_err(|e| handle_mutiny_err(e))?;
+
+    Ok(Json(json!(tx_id.to_string())))
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OpenChannel {
+    amount_sat: u64,
+    fee_rate: Option<f32>,
+    node_pubkey: Option<String>
+}
+
+pub async fn open_channel(
+    Extension(state): Extension<State>,
+    Form(request): Form<OpenChannel>,
+) -> Result<Json<Value>, (StatusCode, Json<Value>)> {
+    let to_pubkey = match request.node_pubkey {
+        Some(pubkey) => {
+            let pubkey = match PublicKey::from_str(pubkey.as_str()) {
+                Ok(pk) => pk,
+                Err(_) => {
+                    return Err((
+                        StatusCode::BAD_REQUEST,
+                        Json(json!({"error": "invalid node pubkey"})),
+                    ))
+                }
+            };
+            Some(pubkey)
+        }
+        None => None
+    };
+
+    let channel = state
+        .mutiny_wallet
+        .node_manager
+        .open_channel(None, to_pubkey, request.amount_sat, request.fee_rate, None)
+        .await
+        .map_err(|e| handle_mutiny_err(e))?;
+    Ok(Json(json!(channel)))
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateInvoiceRequest {
+    amount_sat: u64,
+    label: Option<String>,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateInvoiceResponse {
+    amount_sat: u64,
+    payment_hash: String,
+    serialized: String
+}
+
+pub async fn create_invoice(
+    Extension(state): Extension<State>,
+    Form(request): Form<CreateInvoiceRequest>,
+) -> Result<Json<Value>, (StatusCode, Json<Value>)> {
+    let label = match request.label {
+        Some(label) => vec![label.to_string()],
+        None => Vec::new(),
+    };
+
+    let invoice = state
+        .mutiny_wallet
+        .create_invoice(request.amount_sat, label)
+        .await
+        .map_err(|e| handle_mutiny_err(e))?;
+
+    let amount_sat = match invoice.amount_sats {
+        Some(amount) => amount,
+        None => return Err((
+            StatusCode::INTERNAL_SERVER_ERROR,
+            Json(json!({"error": "unable to create invoice"})),
+        ))
+    };
+
+    let pr = match invoice.bolt11 {
+        Some(amount) => amount,
+        None => return Err((
+            StatusCode::INTERNAL_SERVER_ERROR,
+            Json(json!({"error": "unable to create invoice"})),
+        ))
+    }.to_string();
+
+    let invoice = CreateInvoiceResponse {
+        amount_sat,
+        payment_hash: invoice.payment_hash.to_string(),
+        serialized: pr
+    };
+
+    Ok(Json(json!(invoice)))
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PayInvoiceRequest {
+    amount_sat: Option<u64>,
+    invoice: String
```

Comment:
you can just make this `invoice: Bolt11Invoice` so you don't need to parse it in the function